src/blockchain_db/lmdb/db_lmdb.cpp

```diff
@@ -85,6 +85,36 @@ inline void throw1(const T &e)
 
 #define MDB_val_set(var, val)   MDB_val var = {sizeof(val), (void *)&val}
 
+template<typename T>
+struct MDB_val_nocopy: public MDB_val
+{
+  MDB_val_nocopy(const T &t)
+  {
+    mv_size = sizeof (T);
+    mv_data = &t;
+  }
+};
+
+template<>
+struct MDB_val_nocopy<cryptonote::blobdata>: public MDB_val
```

Comment:
I'd also replace this with a CPP macro, e.g.
#define MDB_val_sized(var, val) MDB_val var = {val.size(), val.data()}